packages/excalidraw/tests/fractionalIndex.test.ts

```diff
@@ -0,0 +1,329 @@
+      movedOneLeftFixedElements,
+      randomlySelected.reduce(
+        (acc, el) => {
+          acc.selectedElementIds[el.id] = true;
+          return acc;
+        },
+        {
+          selectedElementIds: {},
+        } as {
+          selectedElementIds: Record<string, boolean>;
+        },
+      ) as any as AppState,
+    );
+
+    testValidity(movedOneRightFixedElements);
+
+    randomlySelected = [
+      ...new Set([
+        movedOneRightFixedElements[
+          Math.floor(Math.random() * fixedElements.length)
+        ],
+        movedOneRightFixedElements[
+          Math.floor(Math.random() * fixedElements.length)
+        ],
+        movedOneRightFixedElements[
+          Math.floor(Math.random() * fixedElements.length)
+        ],
+        movedOneRightFixedElements[
+          Math.floor(Math.random() * fixedElements.length)
+        ],
+        movedOneRightFixedElements[
+          Math.floor(Math.random() * fixedElements.length)
+        ],
+        movedOneRightFixedElements[
+          Math.floor(Math.random() * fixedElements.length)
+        ],
+        movedOneRightFixedElements[
+          Math.floor(Math.random() * fixedElements.length)
+        ],
+      ]),
+    ];
+
+    const movedAllLeftFixedElements = moveAllLeft(
+      movedOneRightFixedElements,
+      randomlySelected.reduce(
+        (acc, el) => {
+          acc.selectedElementIds[el.id] = true;
+          return acc;
+        },
+        {
+          selectedElementIds: {},
+        } as {
+          selectedElementIds: Record<string, boolean>;
+        },
+      ) as any as AppState,
+    ) as ExcalidrawElement[];
+
+    testValidity(movedAllLeftFixedElements);
+
+    randomlySelected = [
+      ...new Set([
+        movedAllLeftFixedElements[
+          Math.floor(Math.random() * fixedElements.length)
+        ],
+        movedAllLeftFixedElements[
+          Math.floor(Math.random() * fixedElements.length)
+        ],
+        movedAllLeftFixedElements[
+          Math.floor(Math.random() * fixedElements.length)
+        ],
+        movedAllLeftFixedElements[
+          Math.floor(Math.random() * fixedElements.length)
+        ],
+        movedAllLeftFixedElements[
+          Math.floor(Math.random() * fixedElements.length)
+        ],
+        movedAllLeftFixedElements[
+          Math.floor(Math.random() * fixedElements.length)
+        ],
+        movedAllLeftFixedElements[
+          Math.floor(Math.random() * fixedElements.length)
+        ],
+      ]),
+    ];
+
+    const movedAllRightFixedElements = moveAllLeft(
+      movedAllLeftFixedElements,
+      randomlySelected.reduce(
+        (acc, el) => {
+          acc.selectedElementIds[el.id] = true;
+          return acc;
+        },
+        {
+          selectedElementIds: {},
+        } as {
+          selectedElementIds: Record<string, boolean>;
+        },
+      ) as any as AppState,
+    ) as ExcalidrawElement[];
+
+    testValidity(movedAllRightFixedElements);
+  });
```

Comment:
Would also be nice to test breaking ties based `element.id` in case of concurrent updates, followed by the process of fixing conflicting indices and i.e. how it would affect incremental history.

Reply:
This is tested in reconciliation I believe, since `orderByFractionalIndex` is only used there.